### PR TITLE
tests: fix skipped e2e tests

### LINKED ISSUE #5424: Fix E2E tests commented in #5416
PR #5416 commented few tests.

It would be nice to have them working

### PR DESCRIPTION
**Pull request by floyd-li**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

FYI:
1. set `screenshotOnRunFailure` to `true` so we can debug it easier.
 
2. since there's no items in collection `/rmrk/collection/32b209ecbe2b33dd24-DINOXEGGS`, so i just change the collection to `/rmrk/collection/36FDCC8B7888DCED32-SUBWORK_MO` so it can continue the test case.

image

known issues: 

1. i faced several times websocket error like this, and it will break the test case. seems it's a random network issue and there's no rule to reproduce it. 

image

2. the `snekGalleryItem` case failed due to the `unit` display issue on `snek`, is it gonna be fixed in another pr?
reproduce url: `https://beta.kodadot.xyz/snek/gallery/2773267381-1.`

snekGalleryItem -- galleryListedItemActions (failed)

3. the `rmrkGalleryItem` case failed due to a recent update of gql. seems the index need update?
reproduce url: `/rmrk/gallery/9834502-32b209ecbe2b33dd24-DINOXEGGS-DXDANGLS-0000000000000080`

rmrkGalleryItem -- galleryListedItemActions (failed)

image

4. also, other test cases failed sometimes, but it's not related with this pr imo. will look into these tests

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [x] Tests ( I think we need add this PR type :) )

## Context

- [x] Closes #5424
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EZiu1PjV2j2JHKxY6mHnFwwCRCoV27uHKQSkKXATSh1srJT)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at bd31d98</samp>

This pull request adds or updates `data-cy` attributes to various components in the kodadot/nft-gallery project. These attributes are used to locate and interact with elements in the Cypress end-to-end tests. The pull request also enables and refactors some of the existing tests to use the new attributes. The goal is to improve the testability and reliability of the end-to-end tests.

### <samp>🤖 Generated by Copilot at bd31d98</samp>

> _`data-cy` tags_
> _help Cypress find the elements_
> _testing in autumn_

### CODE REVIEW

File: tests/cypress/e2e/basicE2E.cy.ts [hunk 1]
```diff
@@ -15,14 +15,15 @@ describe('landingBsx', () => {
   })
 })
 
-describe.skip('snekCollectionItem', () => {
+describe('snekCollectionItem', () => {
   it('collectionActions', { scrollBehavior: false }, () => {
     cy.visit('/snek/gallery/659233203-5')
-    cy.snekCollectionActions('3132385849', 'Morski pes', 'bXkmyH...J5CSGP')
+    cy.visit('/snek/collection/3132385849')
+    cy.checkCollectionActions('Morski pes', 'bXkmyH...J5CSGP')
```
Comment (Jarsen136) [inline tests/cypress/e2e/basicE2E.cy.ts:22]: I would prefer to put '/snek/collection/3132385849' as a param of `checkCollectionActions` function. In this way, we could visit the related page inside the function.

**Comment by Jarsen136** [comment]
Could you pls also check these failed test cases? Otherwise lgtm

image

**Comment by floyd-li** [comment]
> Could you pls also check these failed test cases? Otherwise lgtm
> 
> image

these cases should be fixed in https://github.com/kodadot/nft-gallery/pull/5794

**Comment by Jarsen136** [comment]
> these cases should be fixed in #5794


Thanks for reminding me. I seem found that there is still a failed case. Let's wait for it to be merged first.

**Comment by Jarsen136** [comment]
Please fix the rest of the test cases and it's good to go.

image

**Comment by floyd-li** [comment]
> Please fix the rest of the test cases and it's good to go.
> 
> image

just checked the screenshot this case i've mentioned in the post. gonna open a issue for it.
image

and this case `Identity.vue component` seems not related with this pr 🤔️, may we fix in another pr?

**Comment by Jarsen136** [comment]
>and this case Identity.vue component seems not related with this pr 🤔️, may we fix in another pr?

Yes

**Comment by Jarsen136** [comment]
> > and this case Identity.vue component seems not related with this pr 🤔️, may we fix in another pr?
> 
> Yes

related with https://github.com/kodadot/nft-gallery/pull/5802

**Review by preschian — APPROVED** [review]

**Review by Jarsen136 — APPROVED** [review]
It works now
image

**Comment by floyd-li** [comment]
@vikiival i think it can be merged  now

**Comment by vikiival** [comment]
Looks like cypress failed?

**Comment by vikiival** [comment]
pay 35 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $35 @ 28.54 USD/KSM ~ 1.226 $KSM
🧗 [EZiu1PjV2j2JHKxY6mHnFwwCRCoV27uHKQSkKXATSh1srJT](https://kusama.subscan.io/account/EZiu1PjV2j2JHKxY6mHnFwwCRCoV27uHKQSkKXATSh1srJT)
🔗 [0x9f275adce219e36c6fb6cd62c37827fafe930b44ba620be0a4294db495b2969d](https://kusama.subscan.io/extrinsic/0x9f275adce219e36c6fb6cd62c37827fafe930b44ba620be0a4294db495b2969d)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**